specification/communication/Communication.Messages/models.tsp

```diff
@@ -374,3 +374,10 @@ enum Versions {
   @useDependency(Azure.Core.Versions.v1_0_Preview_2)
   c2024_02_01: "2024-02-01",
 }
+
+alias BodyParameter<T, name extends valueof string = "body"> = {
+  @friendlyName(name)
+  @bodyRoot
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "Don't want to add documentation"
+  body: T;
+};
```

Comment:
Is there plan to move it to e.g. Azure.Core? Or we just let service add it themselves if they'd like to?

What do we expect new service to do for these POST action APIs? Use the `@bodyRoot` in the Parameter? Write a guide in typespec-azure? Or just let them do the spread as default?

Reply:
This is primarily done for backwards compatibility/not breaking anyone. The default for actions should be positional parameters. I would not expect service teams to use bodyRoot for actions, and if they do want to customize codegen, it would be an augment/client codegen specific override in client.tsp.

Reply:
We will eventually add `BodyParameter` to `Azure.Core`, see this issue: https://github.com/Azure/typespec-azure/issues/1042. There's some design that still needs to happen. Going forward, services can use `BodyParameter`, while languages can use `client.tsp` to override with decorator `overrideClientMethod` (to be added) https://github.com/Azure/typespec-azure/issues/1054

Reply:
Thanks Johan and Isabella.

Yes, moving it to Azure.Core should solve most of the naming ask. Adding one `@@clientName` could make a global change for a language.